htdocs/user/class/user.class.php

```diff
@@ -1417,13 +1417,15 @@ public function create_from_contact($contact, $login = '', $password = '')
 		// Create user and set $this->id. Trigger is disabled because executed later.
 		$result = $this->create($user, 1);
 		if ($result > 0) {
-			$sql = "UPDATE ".MAIN_DB_PREFIX."user";
-			$sql .= " SET fk_socpeople=".$contact->id;
-			$sql .= ", civility=".$contact->civility_code;
+			$sql = 'UPDATE '.MAIN_DB_PREFIX.'user';
+			$sql .= ' SET fk_socpeople='.$contact->id;
+			$sql .= ', civility="'.$contact->civility_code.'"';
```

Comment:
No, it is the opposite. The sql string that must be sent to sql must be
update table set field = 'value' 
and not
update table set field = "value"   (the " is not the standard sql string separator, it is ')

Also you must replace $contact->civility_code with $db->escape($contact->civility_code)